### PR TITLE
auto-random.md: rewrite the document

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: auto-random.md [hunk 1]
```diff
@@ -10,140 +10,90 @@ aliases: ['/docs-cn/dev/auto-random/','/docs-cn/dev/reference/sql/attributes/aut
...
 
-{{< copyable "sql" >}}
-
-```sql
-INSERT INTO t (b) VALUES ("b")
-SELECT * FROM t;
-SELECT last_insert_id()
-```
+- 如果语句中显式指定了 `AUTO_RANDOM` 列的值，则该值会被正常插入到表中；
+- 如果语句中没有显式指定 `AUTO_RANDOM` 列的值，TiDB 会自动生成一个看起来随机的值插入到表中。
```
Comment (bb7133) [inline auto-random.md:32]:
```suggestion
- 如果语句中没有显式指定 `AUTO_RANDOM` 列的值，TiDB 会自动生成一个随机的值插入到表中。
```
我觉得「看起来」可以直接去掉？

**Comment by Oreoxmt** [comment]
@bb7133 PTAL again

File: auto-random.md [hunk 2]
```diff
@@ -12,138 +12,104 @@ aliases: ['/docs-cn/dev/auto-random/','/docs-cn/dev/reference/sql/attributes/aut
...
+| 64 bits | 1 bit  | (64-R) bits | S bits | (R-1-S) bits |
 
-```sql
-+-------+------+----------------------------------------------------------+
-| Level | Code | Message                                                  |
-+-------+------+----------------------------------------------------------+
-| Note  | 1105 | Available implicit allocation times: 1152921504606846976 |
-+-------+------+----------------------------------------------------------+
-```
+- 符号位的长度由该列是否存在 `UNSIGNED` 属性决定：存在则为 `0`，否则为 `1`。
```
Comment (Oreoxmt) [inline auto-random.md:89]:
```suggestion
- 符号位的内容由该列是否存在 `UNSIGNED` 属性决定：存在则为 `0`，否则为 `1`。
```
上表中符号位是 1bit，如果符号位的长度由是否存在 `UNSIGNED` 属性决定，那么上表应该修改为 0/1 bit

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: auto-random.md (hunk 2 above)
Comment (tangenta) [reply auto-random.md:89]: 已修改为 0/1 bit

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: auto-random.md [hunk 3]
```diff
@@ -6,144 +6,112 @@ aliases: ['/docs-cn/dev/auto-random/','/docs-cn/dev/reference/sql/attributes/aut
...
+- 保留位的长度为 `64-R`，保留位的内容始终为 `0`。
+- 分片位的内容通过计算当前事务的开始时间的哈希值而得。要使用不同的分片位数量（例如 10），可以在建表时指定 `AUTO_RANDOM(10)`。
+- 自增位的值保存在存储引擎中，按顺序分配，每次分配完值后会自增 1。自增位保证了 `AUTO_RANDOM` 列值全局唯一。当自增位耗尽后，再次自动分配时会报 `Failed to read auto-increment value from storage engine` 的错误。
 
 > **注意：**
+> 分片位长度 (`S`) 的选取：
```
Comment (Oreoxmt) [inline auto-random.md:91]:
```suggestion
>
> 分片位长度 (`S`) 的选取：
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by ran-huang — APPROVED** [review]

File: auto-random.md [hunk 4]
```diff
@@ -6,144 +6,113 @@ aliases: ['/docs-cn/dev/auto-random/','/docs-cn/dev/reference/sql/attributes/aut
...
-```sql
-CREATE TABLE t (a bigint PRIMARY KEY AUTO_RANDOM)
-```
-
-以上两个语句含义相同。
+> 分片位长度 (`S`) 的选取：
+>
+> - 由于总位数固定为 64 位，分片位的数量会影响到自增位的数量：当分片位数增加时，自增位数会减少，反之亦然。因此，你需要权衡“自动分配值的随机性”以及“可用空间”。
+> - 最佳实践是将分片位设置为 `log(2, x)`，其中 `x` 为当前集群存储引擎的数量。例如，一个 TiDB 集群中存在 16 个 TiKV，分片位可以设置为 `log(2, 16)`，即 `4`。在所有 Region 被均匀调度到各个 TiKV 上以后，此时大批量写入的负载可被均匀分布到不同 TiKV 节点，以实现资源最大化利用。
+> 由于总位数固定为 64 位，分片位的数量会影响到自增位的数量：当分片位数增加时，自增位数会减少，反之亦然。因此，你需要权衡“自动分配值的随机性”以及“可用空间”。
```
Comment (ran-huang) [inline auto-random.md:96]: L94 和 L96 怎么重复了...

**Pull request merged**